x/staking/types/params.go

```diff
@@ -215,17 +215,17 @@ func validateMinCommissionRate(i interface{}) error {
 	return nil
 }
 
-func validateKeyRotationFee(i interface{}) error {
-	v, ok := i.(sdk.Coin)
-	if !ok {
-		return fmt.Errorf("invalid parameter type: %T", i)
+func validateKeyRotationFee(bondDenom string, coin sdk.Coin) error {
+	if strings.TrimSpace(bondDenom) == "" {
+		return errors.New("bond denom cannot be blank")
```

Comment:
May be this check can be removed as we are validating bond denom in line 101 above.

Reply:
good point!